extensions/api/data-management/contractagreement/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractagreement/ContractAgreementApiController.java

```diff
@@ -55,19 +65,27 @@ public ContractAgreementApiController(Monitor monitor) {
                 .sortField(sortField)
                 .filter(filterExpression)
                 .sortOrder(sortOrder).build();
-        monitor.debug(format("get all contract definitions %s", spec));
-
-        return Collections.emptyList();
+        monitor.debug(format("get all contract agreements %s", spec));
 
+        return service.query(spec).stream()
+                .map(it -> transformerRegistry.transform(it, ContractAgreementDto.class))
+                .filter(Result::succeeded)
```

Comment:
We may want to filter out failed transformation results and react to them somehow. This is something we should do in all controllers homogeneously, and not in this PR though.

Reply:
yes, let's address this to another issue